indico/web/forms/fields.py

```diff
@@ -540,3 +541,22 @@ def __init__(self, *args, **kwargs):
 
     def _value(self):
         return self.text_value
+
+
+class IndicoColorSwitchField(JSONField):
+    """Field allowing user to pick a color from a set of predefined values"""
+
+    widget = JinjaWidget('forms/colorswitch_widget.html')
+    default_widget_colors = {'background': 'EEE0EF', 'text': '1D041F'}
+
+    def process_formdata(self, valuelist):
+        super(IndicoColorSwitchField, self).process_formdata(valuelist)
+        self.data = ColorTuple(self.data['text'], self.data['background'])
+
+    def populate_obj(self, obj, name):
+        obj.colors = self.data
+
+    def _value(self):
+        if not self.data:
+            return IndicoColorSwitchField.default_widget_colors
```

Comment:
`self.default_widget_colors`, and I think those should be passed as arguments to the constructor (a decent default value is fine though)